HPCC-21968 Address unwanted logical file popups

miguelvazq opened the pull request:
I have removed the popup warning upon every DFUQueryWidget tab click. I
also noticed an error in the console:

`TypeError: Cannot read property 'Name' of undefined` that I corrected.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

miguelvazq commented:
@kunalaswani can you review and give it a test on an environment with 100,000 + files?

HPCCSmoketest commented:
Process of PR-12503, label: HPCC-21968 starts now.
The reason of this test is: Did not build yet.
Commit ID: 717dbb551589031f1b4de37cfb506cf08c84b1f9
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 717dbb551589031f1b4de37cfb506cf08c84b1f9
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   39 sec (00:00:39) |  177 sec (00:02:57) |    0 sec (00:00:00) |   62 sec (00:01:02) |   21 sec (00:00:21) |    0 sec (00:00:00) |   14 sec (00:00:14) |  313 sec (00:05:13) |

richardkchapman commented:
Missing checkboxes

miguelvazq commented:
FYI @GordonSmith

kunalaswani reviewed (approved):
tested against work-space with more than 100000 files. popup occurs once at initialization and does not after.

richardkchapman commented:
@miguelvazq Please retarget onto candidate-7.0.x

GordonSmith reviewed (approved):
COde changes look ok - but is this meant to target 7.2.x (ins't that where the popup issue happening)?

HPCCSmoketest commented:
Process of PR-12503, label: HPCC-21968 starts now.
The reason of this test is: Base branch changed.
Commit ID: 717dbb551589031f1b4de37cfb506cf08c84b1f9
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 717dbb551589031f1b4de37cfb506cf08c84b1f9
Conflicting files, should skip build and test.

[Build error:](https://gist.github.com/bee24436843251ac0a43d879e1e84f8f)
Build log [Build log](https://gist.github.com/HPCCSmoketest/bee24436843251ac0a43d879e1e84f8f#file-result-19-05-02-08-30-46-log)

miguelvazq commented:
> COde changes look ok - but is this meant to target 7.2.x (ins't that where the popup issue happening)?

Kalaraisi stated it is happening in 7.0.30

Commit:
HPCC-21968 Address unwanted logical file popups

I have removed the popup warning upon every DFUQueryWidget tab click. I
also noticed an error in the console:

TypeError: Cannot read property 'Name' of undefined that I corrected.

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexisrisk.com

miguelvazq commented:
@richardkchapman Retargeted.

HPCCSmoketest commented:
Process of PR-12503, label: HPCC-21968 starts now.
The reason of this test is: New commit.
Commit ID: c359be55593bfe81543f66a263caa0b644c7636f
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: c359be55593bfe81543f66a263caa0b644c7636f
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   35 sec (00:00:35) |  181 sec (00:03:01) |    0 sec (00:00:00) |   60 sec (00:01:00) |   22 sec (00:00:22) |    0 sec (00:00:00) |   15 sec (00:00:15) |  313 sec (00:05:13) |
